autonomica-api/app/services/keyword_clustering.py

```
@@ -0,0 +1,710 @@
+                        model=self.embedding_model
+                    )
+                    
+                    batch_embeddings = [data.embedding for data in response.data]
+                    all_batch_embeddings.extend(batch_embeddings)
+                    
+                    # Rate limiting
+                    await asyncio.sleep(0.1)
+                
+                # Cache new embeddings
+                if use_cache and self.cache_enabled:
+                    for keyword, embedding in zip(uncached_keywords, all_batch_embeddings):
```

Ruff (0.12.2)

`zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)